File: src/main/java/com/lannstark/lec06/Lec06Main.java
package com.lannstark.lec06;

import java.util.Arrays;
import java.util.List;

public class Lec06Main {

  public static void main(String[] args) {

    for (int i = 1; i <= 5; i += 2) {
      System.out.println(i);
    }

    int i = 1;
    while (i <= 3) {
      System.out.println(i);
      i++;
    }

  }

  private void forEach() {
    List<Long> numbers = Arrays.asList(1L, 2L, 3L);
    for (Long number : numbers) {
      System.out.println(number);
    }
  }

  private void traditionalFor() {
    for (int i = 1; i <= 3; i++) {
      System.out.println(i);
    }
  }

  private void traditionalWhile() {
    int i = 1;
    while (i <= 3) {
      System.out.println(i);
      i++;
    }
  }
}

File: src/main/kotlin/com/kimjingyu/lec06/Lec06Main.kt
package com.kimjingyu.lec06

fun main() {
}

fun forEach() {
    val numbers = listOf(1L, 2L, 3L)
    for (number in numbers) {
        println(number)
    }
}

fun traditionFor1() {
    for (i in 1..3) {
        println(i)
    }
}

fun traditionalFor2() {
    for (i in 3 downTo 1) {
        println(i)
    }
}

fun traditionalFor3() {
    for (i in 1..5 step 2) {
        println(i)
    }
}

fun traditionalWhile() {
    var i = 1
    while (i <= 3) {
        println(i)
        i++
    }
}